### PR TITLE
Add "types" declaration to package.json

### PR DESCRIPTION
**Pull request by kirkobyte**
Adds a "types" declaration to the package.json file so TS projects with this as a dependency can find the type declarations

### CODE REVIEW

**Comment by ggdaltoso** [comment]
Hi, @kirkobyte !

First of all, thanks for your contribution!

I'm not a TS expert but why do we need this declaration in the `package.json` of the root?
The published version of `@react95/core` does not have the `package.json` root.

Explaining a little more about the build process, this is what we have:

**@react95/core**: [The first build step](https://github.com/React95/React95/blob/master/packages/core/package.json#L63) will build the publish version of `@react95/core` inside the `dist` folder, containing: `csj`, `esm`, and `@types` folders:

``` 
packages/core/dist
├── @types
├── cjs
└── esm
```

This folder is what you will have in your project (or inside NPM).

After building the package, we run [`prepublishOnly`](https://docs.npmjs.com/cli/v7/using-npm/scripts) which [will generate](https://github.com/React95/React95/blob/master/scripts/prepublish.js) its `package.json` and README.md file.
Since the build process already make the `@types` folder, I do not think it is necessary to have the `@types` field inside `@react95/core` package.json's file:

```
// after running prepublishOnly

packages/core/dist
├── @types
├── README.md
├── cjs
├── esm
└── package.json
```

So, with this in mind, I would like to suggest you do `yarn build:core` on your machine and see if what we have inside  generated `packages/core/dist` folder is enough for you.

**Comment by kirkobyte** [comment]
Hey, thanks for the quick response! Yeah, sorry, I'd tried it out by changing the `package.json` for @react95/core` in my `node_modules`, and didn't correctly pick the `package.json` to modify for the PR. I've fixed that up now :) 

Unfortunately Typescript won't automatically look at the `@types` folder for project dependencies, so currently using `@react95/core` in a Typescript project will result in this error:
image
(I'm using an unmodified Create React App set-up).

As far as I know, `@types` folders are only supported as a default if they're in the root of `node_modules`, as a way of supporting the external typings project, [DefinitleyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). (so users could separately deploy and install `@types/react95__core` if the project was written without types).

Adding `"types": "./@types/",` tells other Typescript projects how to do a type look-up for this dependency. An alternative solution would be to change the build script to output the `.d.ts` definition files to the same location as the `.js` files.

**Comment by ggdaltoso** [comment]
> Unfortunately Typescript won't automatically look at the `@types` folder for project dependencies, so currently using `@react95/core` in a Typescript project will result in this error:
> image
> (I'm using an unmodified Create React App set-up).
> 
> As far as I know, `@types` folders are only supported as a default if they're in the root of `node_modules`, as a way of supporting the external typings project, [DefinitleyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). (so users could separately deploy and install `@types/react95__core` if the project was written without types).

Good to know! Thanks!

File: packages/core/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "Gabriel Daltoso <ggdaltoso@gmail.com>"
   ],
   "license": "MIT",
+  "types": "./@types/",
```
Comment (ggdaltoso) [inline packages/core/package.json:12]:
```suggestion
  "types": "@types",
```

Will this work? 💅

**Review by ggdaltoso — CHANGES_REQUESTED** [review]
Just a typo suggestion, @kirkobyte

Also, I checked your commit messages and it doesn't follow [conventional commit pattern](https://www.conventionalcommits.org/en/v1.0.0/)

We use this pattern to automate our build and publish process, managing which version should be bumped. This is in our [contribute guide](https://github.com/React95/React95/blob/master/CONTRIBUTING.md#directories).

Could you please, revisit your commit messages and edit them?

We have a script to help us with this: `yarn commit`.

To publish a new version, the commit should start with `feat` or `fix`. Although this is a `chore` or `ci` thing, I would suggest you commit as `fix`.

File: packages/core/package.json (hunk 1 above)
Comment (kirkobyte) [reply packages/core/package.json:12]: Yup! Looks like it does :)

**Comment by kirkobyte** [comment]
On a slightly separate note, is there any appetite for migrating `@react95/clippy` to Typescript? I'd be happy to open a PR with the changes!
Also, could you confirm the version of `Node` you use for this project? `.nvmrc` says to use 8, but attempting to run `yarn install` with Node 8 will throw an error stating that Node 10 is expected

**Comment by kirkobyte** [comment]
I've recommitted the changes using `yarn commit` :)

**Comment by ggdaltoso** [comment]
> On a slightly separate note, is there any appetite for migrating @react95/clippy to Typescript? I'd be happy to open a PR with the changes!

It would be awesome to have `@react95/clippy` with TS! Feel free to contribute, man!

> Also, could you confirm the version of Node you use for this project? .nvmrc says to use 8, but attempting to run yarn install with Node 8 will throw an error stating that Node 10 is expected

For this, I don't really know! Would you like to suggest the Node version? I'm really open to suggestions on this... 😂

**Comment by kirkobyte** [comment]
Ah sorry, I used `build`, I can change that to `fix` if you think that's more appropriate.

**Comment by kirkobyte** [comment]
> It would be awesome to have @react95/clippy with TS! Feel free to contribute, man!

Great! I'll have a go at this!

> For this, I don't really know! Would you like to suggest the Node version? I'm really open to suggestions on this... 😂

Node 14 is in Long Term Support mode, so would make a sensible version to use now! It might make sense to change `.nvmrc` and add an `engines` field in `package.json` in a separate PR 😄

**Comment by ggdaltoso** [comment]
> Ah sorry, I used `build`, I can change that to `fix` if you think that's more appropriate.

I was checking it would be a difference between `build` and `fix` but it seems to be none.

Our [`lerna` configuration](https://github.com/React95/React95/blob/master/lerna.json#L8) is skipping changes on `package.json` files and your PR has just it.

If you want to have your PR published, you will need to change [this line](https://github.com/React95/React95/blob/master/lerna.json#L13) too and also make your commit as `fix`

**Comment by kirkobyte** [comment]
Great, fixed :)

**Review by ggdaltoso — APPROVED** [review]

**Comment by ggdaltoso** [comment]
@all-contributors please add @kirkobyte for code

**Comment by naughtyGitCat** [comment]
is there any progress? vs code still hints `npm i --save-dev @types/react95__clippy`

**Comment by ggdaltoso** [comment]
There are no types for @react95/clippy package at the moment, @naughtyGitCat 
But it seems to be easy to fix. 

I'm currently working on a refactor but in the core package. If you can wait, I can do it but it might take some time for me to do this. 

It would be nice if we could have a contribution for you here. What do you think?

**Comment by naughtyGitCat** [comment]
> There are no types for @react95/clippy package at the moment, @naughtyGitCat But it seems to be easy to fix.
> 
> I'm currently working on a refactor but in the core package. If you can wait, I can do it but it might take some time for me to do this.
> 
> It would be nice if we could have a contribution for you here. What do you think?

I would like to do,  but I am newbee in ts, if i have time , i would like to have some try, 😊

**Pull request merged**